precalculus/exercises/outcomes/EQ/EQ2/generator.sage

```diff
@@ -1,10 +1,153 @@
+            "scenarioC",
+            "scenarioD",
+        ])
+        name = choice([
+            "Anne-Fatima",
+            "Boram",
+            "Consolata",
+            "Dalisay",
+            "Edgar",
+            "Fionnuala",
+            "Gurlez",
+            "Hisako",
+            "Inga",
+            "Julia",
+        ])
+
+        data = {}
+
+        if scenario == "distance_toward" or scenario == "distance_apart":
+            d_unit = choice(["mi","km"])
+            vehicles = choice(["trains", "cars", "trucks"])
+            directions = choice([
+                ["north", "south"],
+                ["east", "west"],
+                ["northeast", "southwest"],
+                ["northwest", "southeast"],
+            ])
+            faster_slower = choice(["faster", "slower"])
+            shuffle(directions)
+            mults = sample(range(4,11),2)
+            rates = [20*m for m in mults]
+            time = choice([3,5,7,9,11])/4  # hours
+            dist = sum(rates)*time
+            data = {
+                "dist": dist,
+                "rate1": rates[0],
+                "rate2": rates[1],
+                "rate_diff": abs(rates[0]-rates[1]),
+                "time": time,
+                "d_unit": d_unit,
+                "vehicles": vehicles,
+                "direction1": directions[0],
+                "direction2": directions[1],    
+                "faster_slower": faster_slower,   
+            }
+        
+        if scenario == "mixture_value":
+            part1, part2, percentage = sample([10,20,30,40,60,70,80,90], 3)
+            mixpart = part1*percentage/100 + part2*(100-percentage)/100
+            totalamount = choice(range(10,100,10))
+            amount1 = totalamount*percentage/100
+            amount2 = totalamount*(100-percentage)/100
+
+            if scenario_alt[-1] in ["A", "B"]:
+                weight = choice(["lb", "kg"])
+                ingredient1, ingredient2 = sample([
+                    "chocolate",
+                    "caramel",
+                    "nuts",
+                    "taffy",
+                ], 2)
+            else:
+                weight = choice(["oz"])
+                ingredient1, ingredient2 = sample([
+                    "strawberry",
+                    "banana",
+                    "mango",
+                    "peach",
+                ], 2)
+            data = {
+                "name": name,
+                "weight": weight,
+                "part1": part1,
+                "part2": part2,
+                "percentage": percentage,
+                "mixpart": mixpart,
+                "totalamount": totalamount,
+                "amount1": amount1,
+                "amount2": amount2,
+                "ingredient1": ingredient1,
+                "ingredient2": ingredient2,
+            }
+        
+        if scenario == "mixture_percent":
+            if scenario_alt[-1] in ["A", "B"]:
+                aprs = sample([4,5,6,7,8,9,11,12], 2)
```

Comment:
11 and 12 percent yield seem a bit optimistic--unless you know something I don't, in which case let me know and I'll switch my retirement!

Conversely, 4 and 5 seem a bit low for a loan.  Maybe we should have two sets of aprs, the lower end for the investment and higher for the loans?

Reply:
Sounds fine to me